src/type-layout.md

```diff
@@ -56,12 +56,23 @@ Most primitives are generally aligned to their size, although this is
 platform-specific behavior. In particular, on x86 u64 and f64 are only
 aligned to 32 bits.
 
+For the primitive numeric types (`u8`, `i8`, `u16`, `i16`, `u32`, `i32`, `u64`,
+`i64`, `u128`, `i128`, `usize`, `isize`, `f32`, and `f64`), every bit pattern
+represents a valid instance of the type (in other words,
+`transmute::<[u8; size_of::<T>()], T>(...)` is always sound). For the primitive
+numeric types and also for `bool` and `char`, every byte is guaranteed to be
+initialized (in other words, `transmute::<T, [u8; size_of::<T>()]>(...)` is always
+sound).
```

Comment:
`T` here appears unconstrained and could be misunderstood. Instead it could say something like this:

> For the primitive numeric types `NUM` [...] `transmute::<[u8; size_of::<NUM>()], NUM>(...)`

Reply:
I took a slightly different approach so it'd be consistent across the whole paragraph. How does that look?

Reply:
Well, that leaves the first transmute.

Reply:
I updated the first one too:

> (in other words, for every primitive numeric
type, `T`, `transmute::<[u8; size_of::<T>()], T>(...)` is always sound)

Reply:
Oh yeah, looks fine.